### PR TITLE
Add a note about the widgets binding issue people are reporting

### PR DESCRIPTION
**Pull request by Hixie**
Adds some documentation regarding variants of https://github.com/flutter/flutter/issues/103561.

## Presubmit checklist
- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/master/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

**Review by christopherfujino — APPROVED** [review]
lgtm

File: src/development/tools/sdk/release-notes/release-notes-3.0.0.md [hunk 1]
```diff
@@ -9,6 +9,67 @@ For information about subsequent bug-fix releases, see
 
 [Hotfixes to the Stable Channel]: https://github.com/flutter/flutter/wiki/Hotfixes-to-the-Stable-Channel
 
+## If you see warnings about bindings
+
+You may see warnings like this:
+```
+Warning: Operand of null-aware operation '!' has type 'SchedulerBinding' which excludes null.
+```
+...when migrating to Flutter 3.
```
Comment (domesticmouse) [inline src/development/tools/sdk/release-notes/release-notes-3.0.0.md:18]: @sfshaza2 I'm curious what your preferred way of handling a block of code in the middle of a sentence.

**Review by domesticmouse — APPROVED** [review]
PTAL @sfshaza2

**Comment by Hixie** [comment]
(FWIW, I'm happy to make grammatical fixes in follow-up PRs, but would really like to land this ASAP since it's high-profile.)

**Comment by JacoFourie** [comment]
Dart 2.17 has new lint rules. I think it is those rules that are breaking Flutter 3. I also have this issue. I have been looking around and found this.

https://github.com/dart-lang/lints/blob/main/CHANGELOG.md#200

I think it may be this one

unnecessary_null_aware_assignments

My code will compile if I remove the !. But I have third-party libs that also have them in that I do not have control over. I downgraded my Flutter to 2.10.5 again so I don't have time to look at this now but if somebody can override those lint rules and check if the app compiles.

**Comment by ekuleshov** [comment]
I'm wondering why such backward-incompatible API had been introduced in Flutter 3? It is affecting a lot of plugins and essentially with most of those 3rd party plugins you can't use the same version of plugin between Flutter 2 and Flutter 3.

Wouldn't it be better to keep the old method with existing nullability marked as deprecated in Flutter 3 and have a new method with new nullability added in Flutter 3?

**Pull request merged**